nRF RPC IPC Service no-copy transport (#7510)
nrfconnect/sdk-nrf: doc/nrf/libraries/nrf_rpc/nrf_rpc_ipc.rst

.. _nrf_rpc_ipc_readme:

nRF RPC IPC Service transport
#############################

.. contents::
   :local:
   :depth: 2

The nRF RPC IPC Service transport allows you to use the :ref:`nrf_rpc` library to execute procedures on remote processors.
It uses the IPC Service no-copy mechanism. 

> The nRF RPC IPC Service transport is used in the :ref:`ble_rpc` library and also in the :ref:`nrf_rpc_entropy_nrf53` sample.

Reviewer: ```suggestion
This feature is used in the :ref:`ble_rpc` library and in the :ref:`nrf_rpc_entropy_nrf53` sample.
```
A bet too much repetition if we mention the whole name again. Can we call it a feature?

Author: We can. I changed it

Author: Sure. Done

.. _nrf_rpc_ipc_readme:

nRF RPC IPC Service transport
#############################

.. contents::
   :local:
   :depth: 2

The nRF RPC IPC Service transport allows you to use the :ref:`nrf_rpc` library to execute procedures on remote processors.
It uses the IPC Service no-copy mechanism.

This feature is used in the :ref:`ble_rpc` library and also in the :ref:`nrf_rpc_entropy_nrf53` sample.

Limitations
***********

> 1. The current implementation of the IPC service with the RPMSG backed allows to allocate fixed size buffer.

Reviewer: ```suggestion
1. The current implementation of the IPC service with the RPMSG backend allows to allocate a fixed size buffer.
```
- Should we open the abbreviation RPMSG?
- Use bullet points instead of numbers, as these are not sequential steps.

Author: I find a link to RPMsg so I replaced this sentence in all places with link.
Sure I switched to bullet point list

   Size of this buffer is limited to the:

   .. code-block:: c
      RPMSG_BUFFER_SIZE - sizeof(struct rpmsg_hdr)

> #. The IPC Service function :c:func:`ipc_service_get_tx_buffer` can return after 15 seconds if there is no place for buffer allocation in a shared memory.
>    In case when wait time is set to :c:macro:`K_FOREVER` and the RPMSG backend is used.

Reviewer: ```suggestion
#. The IPC Service function :c:func:`ipc_service_get_tx_buffer` can return after 15 seconds if there is no place for buffer allocation in the shared memory.
   This applies if wait time is set to :c:macro:`K_FOREVER` and the RPMSG backend is used.
```

Author: Done
